app/javascript/store/greetings.js

```diff
@@ -0,0 +1,22 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import axios from 'axios';
+
+const getGreeting = createAsyncThunk('randomGreeting', async () => {
+  const { data } = await axios.get('http://127.0.0.1:3000/api/v1/greetings/random');
+  return data
+});
+
+const greetingSlice = createSlice({
+  name: 'random greeting',
+  initialState: {
+    greeting: { text: 'Hello' },
```

Comment:
**Optional**

- [ ] Also please remember that you are rendering random greetings messages not one message.

Reply:
This is the initial state, it could as well have been an empty object. It is then updated with a random message from the API when the randomMessage component is mounted.

Reply:
yes, correct, but in your case you are not rendering the data stored in the seed, you are calling one object in the initialState.